lib/java/com/google/android/material/appbar/AppBarLayout.java

```diff
@@ -1508,6 +1519,23 @@ public boolean onStartNestedScroll(
       return started;
     }
 
+    private boolean shouldRespectNestedScroll(
+        @NonNull T child,
+        @NonNull View directTargetChild,
+        @NonNull View target) {
+      return (target == directTargetChild && hasScrollingBehavior(target))
```

Comment:
@pubiqq thanks for the PR, can you explain this `target == directTargetChild` check?